### PR TITLE
[LC-525] support clear sign transaction func

### PR DESCRIPTION
**Pull request by StephenC-Ledger**
Update sign transaction func from resolution -> options | Update LedgerEthTransactionResolution to use new parameter/variables

cf.
- https://ledgerhq.atlassian.net/browse/LC-525
- https://github.com/LedgerHQ/ledger-connect/pull/388

### CODE REVIEW

File: Sources/EthWrapper/EthWrapper.swift [hunk 1]
```diff
@@ -91,10 +91,10 @@ public class EthWrapper: BleWrapper {
         }, failure: failure)
     }
     
-    public func signTransaction(path: String, rawTxHex: String, resolution: LedgerEthTransactionResolution?, success: @escaping DictionaryResponse, failure: @escaping ErrorResponse) {
+    public func signTransaction(path: String, rawTxHex: String, option: LedgerEthTransactionResolution?, success: @escaping DictionaryResponse, failure: @escaping ErrorResponse) {
```
Comment (zoltana-ledger) [inline Sources/EthWrapper/EthWrapper.swift]: This is the app you are wrapping and where you are dispatching to https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/hw-app-eth/src/Eth.ts

You can see that signTransaction has a different input, you need to add support for ` async clearSignTransaction(`

The original signTransaction method should not be modified at all.

**Review by zoltana-ledger — CHANGES_REQUESTED** [review]

File: Sources/EthWrapper/ResolutionConfig.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+//  
+//
+//  Created by Stephen Chase on 22/03/2023.
+//
+
+import Foundation
+
+public struct ResolutionConfig: Decodable {
+    public let erc20Tokens: Bool
+    public let nfts: Bool
```
Comment (zoltana-ledger) [inline Sources/EthWrapper/ResolutionConfig.swift]: please tripple check I think the plurals are not correct.

Comment (zoltana-ledger) [reply Sources/EthWrapper/ResolutionConfig.swift]: I'm seeing this:
{ erc20: true, externalPlugins: true, nft: true}

**Review by zoltana-ledger — APPROVED** [review]

File: Sources/EthWrapper/ResolutionConfig.swift (hunk 2 above)
Comment (StephenC-Ledger) [reply Sources/EthWrapper/ResolutionConfig.swift]: Will do, might have referenced the wrong func

**Pull request merged**